adoc/chapters/programming_interface.adoc

```diff
@@ -6057,57 +6067,6 @@ buffer construction.
 ----
 
 
-==== Shared SYCL ownership of the host memory
-
-When an instance of [code]#std::shared_ptr# is passed to the buffer
-constructor, then the buffer object and the developer's application share
-the memory region. If the shared pointer is still used on the application's
-side then the data will be copied back from the buffer or image and will be
-available to the application after the buffer or image is destroyed.
-
-If the [code]#shared_ptr# is not empty, the contents of the referenced
-memory are used to initialize the buffer.  If the [code]#shared_ptr# is
-empty, then the buffer is created with uninitialized memory.
-
-When the buffer is destroyed and the data have potentially been updated, if
-the number of copies of the shared pointer outside the runtime is 0, there
-is no user-side shared pointer to read the data. Therefore the data is not
-copied out, and the buffer destructor does not need to wait for the data
-processes to be finished, as the outcome is not needed on the application's
-side.
-
-This behavior can be overridden using the [code]#set_final_data()#
-member function of the buffer class, which will by any means force the buffer
-destructor to wait until the data is copied to wherever the
-[code]#set_final_data()# member function has put the data (or not wait nor copy
-if set final data is [code]#nullptr)#.
-
-[source,,linenums]
-----
-{
-  std::shared_ptr<int> ptr { data };
-  {
-    buffer<int, 1> b { ptr, range<2>{ 10, 10 } };
-    // update the data
-    [...]
-  } // Data is copied back because there is an user side shared_ptr
-}
-----
-
-[source,,linenums]
-----
-{
-  std::shared_ptr<int> ptr { data };
-  {
-    buffer<int, 1> b { ptr, range<2>{ 10, 10 } };
-    // update the data
-    [...]
-    ptr.reset();
```

Comment:
This line require a comment, for casual readers not using routinely `std::shared_ptr`.